k8.Dockerfile

```diff
@@ -0,0 +1,147 @@
+    tar xzvf /opt/nginx/src/headers-more-nginx-module-0.34.tar.gz -C /opt/nginx/src;
+WORKDIR /opt/nginx/src/nginx-${NGINX_VERSION}
+RUN ./configure --sbin-path=/usr/local/nginx/nginx --prefix=/opt/nginx --with-http_ssl_module --with-ipv6 --with-http_stub_status_module --with-http_realip_module --with-ld-opt="-L/usr/lib/x86_64-linux-gnu" --with-cc-opt="-I/usr/include/x86_64-linux-gnu/openssl" --add-module=/opt/nginx/src/headers-more-nginx-module-0.34 && \
+    make && \
+    make install
+
+RUN ln -s /usr/local/nginx/nginx /usr/local/sbin/nginx; \
+    mkdir -p /opt/nginx/conf/conf.d; \
+    chmod 755 /opt/nginx/conf/conf.d; \
+    mkdir -p /opt/nginx/conf/sites.d; \
+    chmod 755 /opt/nginx/conf/sites.d;
+COPY --chmod=644 ./k8files/status-map.conf /opt/nginx/conf/
+COPY --chmod=644 ./k8files/nginx.conf /opt/nginx/conf/
+COPY --chmod=644 ./k8files/status.conf /opt/nginx/conf/sites.d/
+COPY ./k8files/pivcac.conf /opt/nginx/conf/sites.d/
+
+# Create cron jobs
+RUN echo '* */4 * * * websrv flock -n /tmp/update_cert_revocations.lock -c /usr/local/bin/update_cert_revocations' > /etc/cron.d/update_cert_revocations; \
+    chown root: /etc/cron.d/update_cert_revocations; \
+    chmod 700 /etc/cron.d/update_cert_revocations
+
+# Create working directory
+WORKDIR $RAILS_ROOT
+
+USER app
+
+COPY .ruby-version $RAILS_ROOT/.ruby-version
+COPY Gemfile $RAILS_ROOT/Gemfile
+COPY Gemfile.lock $RAILS_ROOT/Gemfile.lock
+
+RUN bundle config build.nokogiri --use-system-libraries
+RUN bundle config set --local deployment 'true'
+RUN bundle config set --local path $BUNDLE_PATH
+RUN bundle config set --local without 'deploy development doc test'
+RUN bundle install --jobs $(nproc)
+RUN bundle binstubs --all
+
+COPY package.json $RAILS_ROOT/package.json
+
+# Copy Application Code
+COPY --chown=app:app ./lib ./lib
+COPY --chown=app:app ./app ./app
+COPY --chown=app:app ./config ./config
+COPY --chown=app:app ./config.ru ./config.ru
+COPY --chown=app:app ./db ./db
+COPY --chown=app:app ./bin ./bin
+COPY --chown=app:app ./public ./public
+COPY --chown=app:app ./spec ./spec
+COPY --chown=app:app ./vendor ./vendor
+COPY --chown=app:app ./Rakefile ./Rakefile
+COPY --chown=app:app ./Makefile ./Makefile
+COPY --chown=app:app ./Procfile ./Procfile
+COPY --chown=app:app ./log ./log
+COPY --chown=app:app ./tmp ./tmp
+RUN mkdir -p ${RAILS_ROOT}/keys; chmod -R 0755 ${RAILS_ROOT}/keys; \
+    mkdir -p ${RAILS_ROOT}/tmp/cache; chmod -R 0755 ${RAILS_ROOT}/tmp/cache; \
+    mkdir -p ${RAILS_ROOT}/tmp/pids; chmod -R 0755 ${RAILS_ROOT}/tmp/pids; \
+    mkdir -p ${RAILS_ROOT}/tmp/sockets; chmod -R 0755 ${RAILS_ROOT}/tmp/sockets; \
+    mkdir -p ${RAILS_ROOT}/config/puma; chmod -R 0755 ${RAILS_ROOT}/config/puma; 
+COPY --chown=app --chmod=755 ./k8files/application.yml.default.docker ./config/application.yml
+COPY --chown=app --chmod=755 ./k8files/puma_production ./config/puma/production.rb
+
+# Expose port the app runs on
+EXPOSE 443
```

Comment:
We are probably going to need an entrypoint here so that we can make sure that our images can't be run in production by gitlab with arguments that tell it to do random bad things.

Reply:
Added entrypoint and corresponding changes to helm charts in the identity-idp-helm-chart repo